plugins/modules/pnp_intent.py

```diff
@@ -1008,36 +600,95 @@ def get_diff_merged(self):
             class instance for further use.
         """
 
-        if not self.have.get("device_found"):
-            self.log("Adding device to pnp database")
-            response = self.dnac_apply['exec'](
-                family="device_onboarding_pnp",
-                function="add_device",
-                params=self.want.get("pnp_params"),
-                op_modifies=True,
-            )
-
-            self.have["device_id"] = response.get("id")
-            self.log(str(response))
-            self.log(self.have.get("device_id"))
+        device_count_params = {
+            "serial_number": self.want.get("serial_number"),
+            "state": "Provisioned"
+        }
 
-        claim_params = self.get_claim_params()
-        claim_response = self.dnac_apply['exec'](
-            family="device_onboarding_pnp",
-            function='claim_a_device_to_a_site',
-            op_modifies=True,
-            params=claim_params,
-        )
+        if not self.have.get("device_found"):
+            if not self.want["add_device_method"]:
+                self.module.fail_json(msg="Device needs to be added before claiming", response=[])
```

Comment:
set and return instead of failing Json. this should be taken care in check_return_status..

Reply:
Done